### PR TITLE
MySQL 5.7 in Docker even on Ubuntu 20

### PR DESCRIPTION
**Pull request by Hamms**
Install MySQL in Docker in a way that will continue to work after the upgrade to Ubuntu 20; conceptually, this PR mirrors https://github.com/code-dot-org/code-dot-org/pull/53305 but implements the overrides with `debconf-set-selections` rather than by editing apt config files.

## Links

- https://stackoverflow.com/a/37267411/1810460

## Testing story

Relying on existing CI tests to verify that this part of our CI infrastructure continues to work as expected.

Verified that with these changes we can successfully run CI tests on Ubuntu 20 by updating the image we target as base from `ubuntu:18.04` to `ubuntu:20.04`

### CODE REVIEW

File: docker/dockerfiles/Dockerfile [hunk 1]
```diff
@@ -132,22 +133,29 @@ WORKDIR /tmp
...
+# required since it depends on the former.
+#
+# Once we update to MySQL 8+, this can be simplified.
 RUN curl -sSLOJ https://dev.mysql.com/get/mysql-apt-config_0.8.22-1_all.deb && \
     echo "mysql-apt-config mysql-apt-config/select-server select mysql-5.7" | /usr/bin/debconf-set-selections && \
+    echo "mysql-apt-config mysql-apt-config/repo-codename select bionic" | /usr/bin/debconf-set-selections && \
     dpkg -i mysql-apt-config_0.8.22-1_all.deb || apt-get -fy install && \
     rm -rf mysql-apt-config_0.8.22-1_all.deb && \
     apt-get update && \
-    # warning: mysql-server debconf hangs forever if wget is already installed (!)
```
Comment (Hamms) [inline docker/dockerfiles/Dockerfile:141]: Not only does this appear to no longer be true, but without `wget` already being installed, attempting to configure mysql with `dpkg -i mysql-apt-config_0.8.22-1_all.deb` will now fail *unless* `wget` is already installed.

File: docker/dockerfiles/Dockerfile [hunk 2]
```diff
@@ -132,22 +133,29 @@ WORKDIR /tmp
...
+    apt-get install --yes libmysqlclient-dev mysql-client=5.7.42-1ubuntu18.04 mysql-server=5.7.42-1ubuntu18.04
 
 # disable auth for local dashboard mysql db
-RUN chown -R mysql:mysql /var/lib/mysql && \
-    service mysql start && \
+RUN service mysql start && \
     echo "ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY '';" | mysql && \
     echo "CREATE USER 'readonly'@'localhost' IDENTIFIED WITH mysql_native_password BY '';" | mysql && \
     echo "GRANT SELECT ON dashboard_test.* TO 'readonly'@'localhost';" | mysql && \
-    chown -R mysql:mysql /var/lib/mysql /var/run/mysqld && \
```
Comment (Hamms) [inline docker/dockerfiles/Dockerfile:150]: It's not clear to me why we were doing this before (nor, it seems, [was it clear to the last folks to find themselves here](https://github.com/code-dot-org/code-dot-org/pull/29503#discussion_r302259709)), but it no longer seems to be required and starting with Ubuntu 20 it causes some problems. So, I propose we remove it for the same reason as we added it in the first place: doing so make this work.

**Review by cat5inthecradle — APPROVED** [review]

File: docker/dockerfiles/Dockerfile [hunk 3]
```diff
@@ -132,22 +133,29 @@ WORKDIR /tmp
...
+# Install MySQL.
+#
+# We explicitly target MySQL 5.7, and because of that we also explicitly target
+# Ubuntu Bionic (18.x).
+#
+# We also explicitly install both mysql-client and mysql-server, so that we can
+# also explicitly define the versions of both; normally only the latter is
+# required since it depends on the former.
+#
+# Once we update to MySQL 8+, this can be simplified.
```
Comment (cat5inthecradle) [inline docker/dockerfiles/Dockerfile:145]: I appreciate that this comment will be discoverable by anyone searching for "mysql 5.7" or "ubuntu". Not sure if you've been intentional about that in all these PR's, but if so, thanks!

Comment (Hamms) [reply docker/dockerfiles/Dockerfile:145]: Yeah! I've learned through doing all these upgrades that explicit library names and version numbers in comments are super valuable, particularly in future-facing comments like this one.

**Pull request merged**